contracts/TokenFactory.sol

```diff
@@ -305,64 +305,35 @@ contract TokenFactory is ProxyFactory, Ownable, ITokenFactory {
         return false;
     }
 
-    function mint721(
+    function mintBatch721(
         address nft,
         address to,
-        uint256 tokenId,
-        bytes memory data,
+        uint256[] calldata tokenIds,
+        bytes calldata data,
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) public override {
+    ) external override {
         address owner = IBaseNFT721(nft).owner();
-        bytes32 hash = keccak256(abi.encode(NFT721_TYPEHASH, nft, to, tokenId, data, nonces721[owner]++));
+        bytes32 hash = keccak256(abi.encode(NFT721_TYPEHASH, nft, to, tokenIds, data, nonces721[owner]++));
         Signature.verify(hash, owner, v, r, s, DOMAIN_SEPARATOR());
-        IBaseNFT721(nft).mint(to, tokenId, data);
+        IBaseNFT721(nft).mintBatch(to, tokenIds, data);
     }
 
-    function mint1155(
+    function mintBatch1155(
         address nft,
         address to,
-        uint256 tokenId,
-        uint256 amount,
-        bytes memory data,
+        uint256[] calldata tokenIds,
+        uint256[] calldata amounts,
+        bytes calldata data,
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) public override {
+    ) external override {
         address owner = IBaseNFT1155(nft).owner();
-        bytes32 hash = keccak256(abi.encode(NFT1155_TYPEHASH, nft, to, tokenId, amount, data, nonces1155[owner]++));
+        bytes32 hash = keccak256(abi.encode(NFT1155_TYPEHASH, nft, to, tokenIds, amounts, data, nonces1155[owner]++));
         Signature.verify(hash, owner, v, r, s, DOMAIN_SEPARATOR());
-        IBaseNFT1155(nft).mint(to, tokenId, amount, data);
-    }
-
-    function mintWithTags721(
-        address nft,
-        address to,
-        uint256 tokenId,
-        bytes memory data,
-        string[] memory tags,
-        uint8 v,
-        bytes32 r,
-        bytes32 s
-    ) external override {
-        mint721(nft, to, tokenId, data, v, r, s);
-        _setTags(nft, tokenId, tags);
-    }
-
-    function mintWithTags1155(
-        address nft,
-        address to,
-        uint256 tokenId,
-        uint256 amount,
-        bytes memory data,
-        string[] memory tags,
-        uint8 v,
-        bytes32 r,
-        bytes32 s
-    ) external override {
-        mint1155(nft, to, tokenId, amount, data, v, r, s);
-        _setTags(nft, tokenId, tags);
+        IBaseNFT1155(nft).mintBatch(to, tokenIds, amounts, data);
```

Comment:
https://github.com/sushiswap/shoyu/blob/master/contracts/base/BaseNFT1155.sol#L106
https://github.com/sushiswap/shoyu/blob/master/contracts/base/BaseNFT1155.sol#L117

The require conditions should be swapped.